MUXSDKImaKalturaListener/Sources/MUXSDKImaKalturaListener.swift

```diff
@@ -0,0 +1,95 @@
+//
+//  MUXSDKImaKalturaListener.swift
+//  Mux-Stats-Google-IMA-Kaltura
+//
+//  Created by Stephanie Zuñiga on 27/1/22.
+//
+
+import MUXSDKKaltura
+import MuxCore
+import PlayKit
+import GoogleInteractiveMediaAds
+
+public class MUXSDKImaKalturaListener {
+    private let dispatcher: MUXSDKImaKalturaDispatcher
+    private let player: Player
+    
+    public init(playerBinding: MUXSDKPlayerBinding, player: Player) {
+        var isTesting: Bool {
+            return ProcessInfo.processInfo.arguments.contains("TEST")
+        }
+        
+        self.dispatcher = isTesting ? MockedDispatcher.shared : MUXSDKImaKalturaMainDispatcher(playerBinding: playerBinding)
+        self.player = player
+    }
+    
+    public func start() {
+        let events: [PKEvent.Type] = [
+            AdEvent.adDidRequestContentPause,
+            AdEvent.adDidRequestContentResume,
+            AdEvent.adLoaded,
+            AdEvent.adStarted,
+            AdEvent.adFirstQuartile,
+            AdEvent.adMidpoint,
+            AdEvent.adThirdQuartile,
+            AdEvent.adSkipped,
+            AdEvent.adComplete,
+            AdEvent.adPaused,
+            AdEvent.adResumed,
+            AdEvent.error
+        ]
+        
+        self.player.addObserver(self, events: events) { event in
+            self.dispatch(event: event)
+        }
+    }
+    
+    private func dispatch(event: PKEvent) {
+        var playbackEvents = [MUXSDKPlaybackEvent]()
+        
+        switch event {
+        case is AdEvent.AdDidRequestContentPause:
+            playbackEvents.append(MUXSDKAdBreakStartEvent())
+            playbackEvents.append(MUXSDKAdPlayEvent())
+        case is AdEvent.AdDidRequestContentResume:
+            playbackEvents.append(MUXSDKAdBreakEndEvent())
+        case is AdEvent.AdLoaded:
+            playbackEvents.append(MUXSDKAdRequestEvent())
```

Comment:
I think we shouldn't map them both in AdLoaded bc it makes it seem like they happened at the same exact time. I liked the solution you had previously where you used the adRequested. Even though we will only get one adRequest in that case it seems more correct? 🤔  

It produced a view like this https://dashboard.mux.com/organizations/72polf/environments/u2qvlr/views/LrGJnBNig4bL0Whl9Wc9LDFq81HxPN9le2GG

Reply:
Yeah sorry, I thought we wanted to keep the logic closer to what we have in avPlayer which maps a request event for each ad. However, I agree with you it's more correct to map the ads requested event from kaltura into our mux adRequest event and then the adLoaded for the responses cause playkit-ima requests the stream of ads once and then it generates and adLoaded event for each ad response. 

I updated the logic and generated a new view with it, let me know if it's aligned with what we want to track. 
https://dashboard.mux.com/organizations/72polf/environments/u2qvlr/views/3RAWpbBtmVPz5aHdNMc55juMBksroPnYE9a1